src/topology/algebra/ordered.lean

```diff
@@ -721,6 +721,31 @@ induced_order_topology' f @hf
   (λ a x xa, let ⟨b, xb, ba⟩ := H xa in ⟨b, hf.1 ba, le_of_lt xb⟩)
   (λ a x ax, let ⟨b, ab, bx⟩ := H ax in ⟨b, hf.1 ab, le_of_lt bx⟩)
 
+/-- On an open subinterval of a dense linear order, the order topology for the restriction of the
+order is the same as the restriction to the interval of the order topology. -/
+instance {α : Type*}
+  [topological_space α] [decidable_linear_order α] [densely_ordered α] [order_topology α]
+  {a b : α} :
+  order_topology (Ioo a b) :=
```

Comment:
This is presumably also true for any set `s` with `is_interval s`?

Reply:
I think the lemma I use to prove this, `induced_order_topology'`, wouldn't apply for (eg) `s = Icc`.  One of the hypotheses translates (please check this?) to, `∀ {x y : α}, x ∈ s → y < x → ∃ {z : α}, z ∈ s ∧ z < x ∧ y ≤ z`. This would fail for `s = Icc a b` with `x = a`.

Of course, it might well be true by a different method.

Reply:
I think the result is true for any `ord_connected` set. If you can't find a proof, tell me and I'll have a look!

Reply:
This works!  (Slightly long proof, golfing welcome.)